misc/python/materialize/mzcompose/__init__.py

```diff
@@ -220,8 +220,17 @@ def __init__(
                     name = name[len("workflow_") :].replace("_", "-")
                     self.workflows[name] = fn
 
+            seen_services = set()
             for python_service in getattr(module, "SERVICES", []):
-                compose["services"][python_service.name] = python_service.config
+                name = python_service.name
+                if name in seen_services:
+                    raise UIError(f"Service specified twice in mzcompose.py: {name}")
+                seen_services.add(name)
+                if name in compose["services"]:
+                    raise UIError(
+                        f"Service specified in both mzcompose.py and mzcompose.yml: {name}"
+                    )
```

Comment:
I'm confused about the existence of `seen_services`. Can we instead just keep this `if` statement and say "service specified twice"? mzcompose.yml will be going away so it doesn't seem worth adding extra complexity for the sake of an error message.

Reply:
yeah, I added the extra thing specifically to make the error message better, since you don't want that I'll get rid of it.

Reply:
Yeah, strong preference for simplicity.